feat: file structure section created

drake-stamos opened the pull request.

arcanaxion committed:
Fix "multiple" typo

arcanaxion committed:
Removed "write 1 row" examples

arcanaxion committed:
Grammar updates

arcanaxion committed:
Remove "previous section" phrase

drake-stamos committed:
feat: file structure section created

drake-stamos committed:
feat: file structure section created

haotran-avaiga committed:
Add document for funnel charts (#306)

* Add funnel chart doc section

haotran-avaiga committed:
Create a new doc section for filled area plots (#293)

* Add filled area chart section
* including Normalized Stacked Area Chart

drake-stamos committed:
Add Polar chart (#284, #259) (#294)

* Added polar chart section and examples
* Conventions and additions.

haotran-avaiga committed:
Create a heatmap charts section (#286) (#292)

* Add a document and images for heatmap chart
* Add examples

FabienLelaquais committed:
Integrate local html template for (#334)

Relocate charts home page.

dinhlongviolin1 commented on docs/manuals/gui/extension_file_structure.md:
````diff
@@ -0,0 +1,92 @@
+    
+      - `index.ts` is where all react elements gets exported with the name that matches with our `demo_library.py` elements' name.
+    ```ts
+      // index.ts
````
I am thinking whether we should keep a demo right here

drake-stamos replied:
@dinhlongviolin1 Do you have any idea?

dinhlongviolin1 commented on docs/manuals/gui/extension_file_structure.md:
````diff
@@ -0,0 +1,92 @@
+            return ["demo_lib/frontend/dist/demo.js"]
+    ```
+- `demo-init.py` is the entry point for our library. This is where we use our taipy elements
+```py
````
same thing

FabienLelaquais committed:
Create the bubble charts section (#335)

Update and conventions.

haotran-avaiga committed:
Add radar charts (#305) (#311)

* Add radar charts section with examples.

FabienLelaquais committed:
Create Pie Charts section (#336) (#337)

* New section created.
* Examples refined

drake-stamos committed:
doc: added react code

drake-stamos committed:
doc: added react code

FabienLelaquais committed:
Create the Error Bars section (#341). (#342)

* Created the error bars section.

Commit:
feat(home page): Add "Charts" to quick access and relayout quick access grid.

Commit:
feat(manuals/about): Add quick access cards to user manual landiing page

Commit:
feat(style): Replace "glowing logo" with clean white logo

drake-stamos committed:
Add a Candlestick charts section (#285) (#297)

* Added candlestick charts section

trgiangdo committed:
feat: add documentation for versioning system

trgiangdo committed:
fix: remove click from pipfile

trgiangdo committed:
fix: simplify the code example on index

trgiangdo committed:
fix: simplify the code example on console example

trgiangdo committed:
feat: add legacy app migration page

drake-stamos committed:
Add a section for continuous error charts (#303) (#315)

New section for Continuous Error charts

jrobinAV committed:
release notes

jrobinAV committed:
release notes

joaoandre-avaiga committed:
chore: add examples for generic data node

joaoandre-avaiga committed:
chore: remove extra blank lines

haotran-avaiga committed:
New section on plotting on maps (#304) (#310)

Add doc section for plotting on maps (lines & bubbles)

arcanaxion committed:
Doc/#276 ParquetDataNode documentation (#298)

Co-authored-by: Đỗ Trường Giang <dtr.giang.1299@gmail.com>
Co-authored-by: Jean-Robin <jeanrobin.medori@avaiga.com>

jrobinAV committed:
Merge pull request #322 from arcanaxion/doc/#321-dn-mgt-cfg-updates

* Fix "multiple" typo
* Removed "write 1 row" examples
* Grammar updates
* Remove "previous section" phrase

jrobinAV committed:
Merge branch 'develop' into feature/doc#339

jrobinAV committed:
Fix generic data node code sample + minor changes + reformatting

joaoandre-avaiga committed:
Merge pull request #348 from Avaiga/feature/doc#339

add examples for generic data node

jrobinAV committed:
update page names

jrobinAV committed:
update Version management page

jrobinAV committed:
update Manage page:
 - add option s description
 - add help example
 - simplify main.py

jrobinAV committed:
update Development version with more details and simpler example

jrobinAV committed:
Merge branch 'develop' into feat/#340-versioning-doc

# Conflicts:
#	docs/relnotes.md

FabienLelaquais committed:
New Basic and Advanced content (#345) (#350)

* New Basic and Advanced content.
Co-authored-by: Fabien Lelaquais <fabien@lelaquais.fr>
Co-authored-by: Fred Lefévère-Laoide <90181748+FredLL-Avaiga@users.noreply.github.com>

jrobinAV committed:
Update Experiment version with more details and simpler examples

jrobinAV committed:
Update Production version with more details and simpler examples

jrobinAV committed:
Update migration path page.

jrobinAV committed:
remove unused python files

jrobinAV committed:
polishing

jrobinAV committed:
Update docs/manuals/core/versioning/code_example/main.py

Co-authored-by: Đỗ Trường Giang <do.giang@avaiga.com>

jrobinAV committed:
Update docs/manuals/core/versioning/code_example/main_with_changes.py

Co-authored-by: Đỗ Trường Giang <do.giang@avaiga.com>

jrobinAV committed:
Update docs/manuals/core/versioning/experiment_mode.md

Co-authored-by: Đỗ Trường Giang <do.giang@avaiga.com>

jrobinAV committed:
Update docs/manuals/core/versioning/experiment_mode.md

Co-authored-by: Đỗ Trường Giang <do.giang@avaiga.com>

jrobinAV committed:
Update docs/manuals/core/versioning/experiment_mode.md

Co-authored-by: Đỗ Trường Giang <do.giang@avaiga.com>

jrobinAV committed:
Update docs/manuals/core/versioning/experiment_mode.md

Co-authored-by: Đỗ Trường Giang <do.giang@avaiga.com>

jrobinAV committed:
Merge remote-tracking branch 'origin/feat/#340-versioning-doc' into feat/#340-versioning-doc

jrobinAV committed:
add print in main files

FabienLelaquais committed:
Create a new section in the User Manual for Taipy Studio (#351)

Initiate Taipy Studio user manual section

jrobinAV committed:
Merge branch 'develop' into feat/#340-versioning-doc

jrobinAV committed:
Merge pull request #346 from Avaiga/feat/#340-versioning-doc

Feat/#340 - Taipy Core version control documentation

Commit:
Use npm 9.4

jrobinAV committed:
replace cacheable by skippable

jrobinAV committed:
Merge pull request #352 from Avaiga/feature/#398-cacheable

replace cacheable by skippable

jrobinAV committed:
upgrades page

jrobinAV committed:
upgrades page

(cherry picked from commit a4e7496d1d2d0730da6b18aeeafa5f7fdfb90440)

jrobinAV committed:
Revert "upgrades page"

This reverts commit a4e7496d1d2d0730da6b18aeeafa5f7fdfb90440.

FabienLelaquais committed:
Updated Gantt chart doc - new example (#353)

FabienLelaquais committed:
Simplify Gantt chart doc (#245) (#355)

* Simplified thanks to Fred's insistence

jrobinAV committed:
Vincent feedback: Concepts

jrobinAV committed:
Merge remote-tracking branch 'origin/develop' into develop

jrobinAV committed:
Vincent feedback: core/configuration

jrobinAV committed:
FIx wrong file filtering when fetching source code

jrobinAV committed:
Vincent feedback : core/Entities

FabienLelaquais committed:
Documentation for Taipy Studio GUI (#318) (#358)

Start Taipy Studio GUI user man section

FabienLelaquais committed:
Update Release Notes for Taipy GUI 2.1.0 (#356) (#357)

Update Release Notes for Taipy GUI 2.1.0

jrobinAV committed:
Merge branch 'develop' into feature/#448-upgrades

jrobinAV committed:
Include some remarks from PR review

jrobinAV committed:
Include all remarks from PR review

jrobinAV committed:
Merge pull request #354 from Avaiga/feature/#448-upgrades

migration page

jrobinAV committed:
Vincent feedback: Versioning management system

jrobinAV committed:
simplify versioning/development-mode

jrobinAV committed:
minor typo on message

jrobinAV committed:
minor refactoring of Experiment mode

jrobinAV committed:
SQL repository in release notes

jrobinAV committed:
SQL repository in release notes

jrobinAV committed:
Propose core quick access buttons

Commit:
Merge pull request #343 from Avaiga/feat/add-charts-quick-access

Feature(home page): Add "charts" to quick access

jrobinAV committed:
Merge branch 'develop' into feature/#338-core-quick-access

jrobinAV committed:
improve self-explanatory pills

jrobinAV committed:
remove tp-tooltip

jrobinAV committed:
Merge pull request #362 from Avaiga/feature/#338-core-quick-access

Propose core quick access buttons

jrobinAV committed:
Vincent feedback: advanced-config

jrobinAV committed:
FIx tom config in advanced-config

FabienLelaquais committed:
Documentation for Taipy Studio Configuration Builder (#319) (#364)

* Doc on Studio Config Builder

Commit:
Release 2.1

jrobinAV committed:
add rel notes entry for overload methods

FabienLelaquais committed:
Update issue templates

jrobinAV committed:
Update mkdocs.yml_template

jrobinAV committed:
Feature/getting started (#367)

* fetch_source_file refacto: extract GitContext

* fetch_source_file refacto: extract CLI

* fetch_source_file refacto: extract read version from mkdocs.yml

* fetch new taipy-getting-started for gui and core

* set_up_generation for new getting started gui and core

* improve step_getting_started

* improve step_getting_started

* Update mkdocs.yml_template

* Create landing page for Getting Started

* Update index.md

* Update index.md

* Fix small issues

* feedback from PR review

* remove print

* Update docs/getting_started/index.md

* Update docs/getting_started/index.md

* use double quotes

---------

Co-authored-by: FlorianJacta <98709993+FlorianJacta@users.noreply.github.com>

jrobinAV committed:
add missing import

jrobinAV committed:
attempt to fix commit empty directory

jrobinAV committed:
attempt to fix commit empty directory

trgiangdo committed:
fix: config conflict typo (#370)

jrobinAV committed:
Merge pull request #369 from Avaiga/feature/overload-methods

Feature/overload methods

jrobinAV committed:
refman: core + core.common

jrobinAV committed:
refman: core + core.common

jrobinAV committed:
refman: core + core.common

jrobinAV committed:
minor changes

jrobinAV committed:
minor changes

jrobinAV committed:
expose DefaultCustomDocument at the right level

jrobinAV committed:
fix broken link on tai^py enterprise package ref manual

(cherry picked from commit 10d6d9eb92b7543b8a53419ee21fd60f56cf4df7)

jrobinAV committed:
Convert and reuse hack from fetch_source.sh to add taipy.run method into init.py file

(cherry picked from commit ebd942adbd94f2f4347c742cda9246cea2b24b48)

jrobinAV committed:
fix generated config.py

(cherry picked from commit e20034b415e520f0686fa706337952cfae33c242)

FabienLelaquais committed:
Taipy Core type aliases (#376) (#378)

- Bring Taipy Core type alises back
- Minor mkdocs warning fixed
- move aliases to the right location taipy.core

Co-authored-by: Fabien Lelaquais <fabien@lelaquais.fr>
Co-authored-by: jrobinAV <88036007+jrobinAV@users.noreply.github.com>

FabienLelaquais committed:
Complement Taipy Studio's documentation  with the Graph View (#366) (#377)

- Complemented with the Graph View
- mkdocs template
- Animations for Graph View edition
- Remarks integrated

FabienLelaquais committed:
Direct links to chart code examples in taipy-gui (#379) (#383)

- Converts data-source HTML attributes to a link to a file within a Taipy repository
- Update missed copyright dates.

jrobinAV committed:
attempt to fix doc

jrobinAV committed:
attempt 2 to fix doc

jrobinAV committed:
attempt 3 to fix doc

jrobinAV committed:
attempt 4 to fix doc

jrobinAV committed:
attempt 5 to fix doc

jrobinAV committed:
attempt 6 to fix doc

jrobinAV committed:
attempt 7 to fix doc

jrobinAV committed:
attempt 8 to fix doc : Manual removal of overload

jrobinAV committed:
Add comments

Commit:
added warning about reserved keywords

Commit:
fix changing layout due to extension

Commit:
fix changing layout due to extension for relnote

Commit:
fix changing layout due to extension for scenario_config.md

Commit:
fix changing layout due to extension for data_config.md

Commit:
fix changing layout due to extension for data_config.md

Commit:
fix changing layout due to extension for data_config.md

jrobinAV committed:
Add comments

Commit:
added warning titles and remove modin version change to a different pr

Commit:
added doc for changes in openpyxl version

toan-quach committed:
Merge pull request #388 from Avaiga/feature/#443-cannot-use-parent-as-property-key

feature/#443 added warning about reserved keywords

toan-quach committed:
Merge pull request #389 from Avaiga/bug/#473-add-note-openpyxl-version-change-relnotes

bug/#473 added doc for changes in openpyxl version

Commit:
Merge branch 'doc/extension-file-structure' of github.com:Avaiga/taipy-doc into doc/extension-file-structure

Commit:
Extension library introduction exposes the project structure

FredLL-Avaiga commented on docs/manuals/gui/extension/index.md:
```diff
@@ -0,0 +1,212 @@
+- An element library is a Python object that derives from the `ElementLibrary^` class. 
+- Element libraries have a name (returned by `ElementLibrary.get_name()^`) that is
+  used to refer to visual elements in page definitions.
+- Element libraries hold the set of its elements, as returned by
```
its => theirs ?

FabienLelaquais replied:
Correct! (well 'their')
Thx

FredLL-Avaiga commented on docs/manuals/gui/extension/index.md:
```diff
@@ -0,0 +1,212 @@
+- Static elements.<br/>
+  A static visual element cannot interact with the underlying application.
+  It can be used just like any other element in a page, but its properties are not
+  bound to the application variable: if the variable value is modified, it does not
```
application variables ?
If a variable value is

FabienLelaquais replied:
Correct!

FredLL-Avaiga commented on docs/manuals/gui/extension/index.md:
```diff
@@ -0,0 +1,212 @@
+
+- Taipy GUI 2.0 or higher (included in Taipy and Taipy Enterprise).
+- Python 3.8 or higher.
+- NodeJS (and NPM) if you need to create dynamic custom visual elements.
```
minimal version of NodeJs ? 18 ?

FabienLelaquais replied:
18, ok.
I was trying to avoid saying it in case it changes.

FredLL-Avaiga commented on docs/manuals/gui/extension/index.md:
```diff
@@ -0,0 +1,212 @@
+dynamic elements is accessible with its entire source code and build process in
+the `doc/extension` directory under the root directory of the Taipy GUI installation.<br/>
+You can also take a look at this extension library example directly on
+[GitHub](https://github.com/Avaiga/taipy-gui/tree/[BRANCH]/doc/extension).
```
do you replace [BRANCH] ?

FabienLelaquais replied:
I do

Commit:
Comments.

Commit:
Comments

Commit:
Comments
